miner/worker.go

```diff
@@ -376,6 +395,7 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 		}
 		timer.Reset(recommit)
 		atomic.StoreInt32(&w.newTxs, 0)
+		atomic.StoreInt32(&w.newL1Msgs, 0)
```

Comment:
The logic here might be wrong because a new block may not be able to include all l1 msgs due to the L1 msg limit.

Reply:
Even though we maintain a count here, we only use it to check if there were any new messages (`atomic.LoadInt32(&w.newL1Msgs) == 0`), the actual count is not used. `newL1Msgs` follows the same logic as `newTxs`.

Reply:
Yes, but the issue is that if there are still pending L1 messages in the queue after the new block generated, it will no longer trigger a new block unless there are L1 msgs in the new L1 blocks or new L2 txs.

Reply:
I think you're right. Here I just copied the logic of `newTxs` which then suffer from the same issue. I'm not sure it's a big issue in practice though, because
1. We'll disable the "no empty blocks" feature,
2. Even if we still disallowed empty blocks, the frequency of incoming L1 and L2 txs is expected to continuously trigger new blocks.